provider/pkg/await/deployment.go

```diff
@@ -144,53 +146,88 @@ func (dia *deploymentInitAwaiter) Await() error {
 	//      because it doesn't do a rollout (i.e., it simply creates the Deployment and
 	//      corresponding ReplicaSet), and therefore there is no rollout to mark as "Progressing".
 	//
-
-	deploymentClient, replicaSetClient, podClient, pvcClient, err := dia.makeClients()
-	if err != nil {
-		return err
-	}
-
-	// Create Deployment watcher.
-	deploymentWatcher, err := deploymentClient.Watch(context.TODO(), metav1.ListOptions{})
-	if err != nil {
-		return errors.Wrapf(err, "could not set up watch for Deployment object %q",
-			dia.config.currentInputs.GetName())
-	}
-	defer deploymentWatcher.Stop()
-
-	// Create ReplicaSet watcher.
-	replicaSetWatcher, err := replicaSetClient.Watch(context.TODO(), metav1.ListOptions{})
-	if err != nil {
-		return errors.Wrapf(err,
-			"Could not create watcher for ReplicaSet objects associated with Deployment %q",
-			dia.config.currentInputs.GetName())
-	}
-	defer replicaSetWatcher.Stop()
-
-	// Create Pod watcher.
-	podWatcher, err := podClient.Watch(context.TODO(), metav1.ListOptions{})
-	if err != nil {
-		return errors.Wrapf(err,
-			"Could not create watcher for Pods objects associated with Deployment %q",
-			dia.config.currentInputs.GetName())
-	}
-	defer podWatcher.Stop()
-
-	// Create PersistentVolumeClaims watcher.
-	pvcWatcher, err := pvcClient.Watch(context.TODO(), metav1.ListOptions{})
-	if err != nil {
-		return errors.Wrapf(err,
-			"Could not create watcher for PersistentVolumeClaims objects associated with Deployment %q",
-			dia.config.currentInputs.GetName())
-	}
-	defer pvcWatcher.Stop()
+	stopper := make(chan struct{})
+	defer close(stopper)
+
+	namespace := dia.deployment.GetNamespace()
+	if namespace == "" {
+		namespace = metav1.NamespaceDefault
+	}
+	informerFactory := dynamicinformer.NewFilteredDynamicSharedInformerFactory(dia.config.clientSet.GenericClient, 60*time.Second, namespace, nil)
+	// Limit the lifetime of this to each deployment await for now. We can reduce this sharing further later.
+	informerFactory.Start(stopper)
+
+	deploymentEvents := make(chan watch.Event)
+	deploymentV1Informer := dia.makeInformer(
+		informerFactory,
+		schema.GroupVersionResource{
+			Group:    "apps",
+			Version:  "v1",
+			Resource: "deployments",
+		}, deploymentEvents)
+	go deploymentV1Informer.Informer().Run(stopper)
+	deploymentV1Beta1Informer := dia.makeInformer(
+		informerFactory,
+		schema.GroupVersionResource{
+			Group:    "extensions",
+			Version:  "v1beta1",
+			Resource: "deployments",
+		}, deploymentEvents)
+	go deploymentV1Beta1Informer.Informer().Run(stopper)
```

Comment:
I'm not sure if we need Informers for the old apiVersions. I believe the watch clients were previously only using the latest apiVersions.

Reply:
Ah interesting. Does that mean that we don't really support the v1beta1 etc. variants? It seems like we load the latest api versions when creating clients and use them to create watches. I can't really verify this since all the cloud providers seem to have stopped supporting  1.15 or older (bunch of these were killed in 1.16).

Reply:
Caught up with @lblackstone offline. It seems we should be safe here. Removed non-v1 informer variants.